Utrecht feature/domain

Noor-Serry opened the pull request:
add usedNumber field to Category and add getTasksByCategoryAndStatus to TaskRepository
also I added the repository to store app setting like current theme mode and is onboarding screen is opend or not yet 
 - then I rename the usedNumber field to number of tasks also I rename the function name to getTasksByCategoryIdAndStatus 
 - add appSettingsRepository with four function to track if the user opend onboarding screen is opend or not yet also to track the selected user theme
 - create theme enum with two field Light and Dark

ThoriaHamdy99 committed:
define cute tudee app colors.

ThoriaHamdy99 committed:
add dark theme colors.

ThoriaHamdy99 committed:
move the app colors to colors package under theme package.

Ahmed-AlMancy committed:
add nunito font family to the font resource file

Ahmed-AlMancy committed:
add data classes for both textStyle and their sizes

Ahmed-AlMancy committed:
add the default textStyle for their sizes

Ahmed-AlMancy committed:
add font xml file

Ahmed-AlMancy committed:
fix fontSize and height for title large size

mohamedd-hassan committed:
added string resources

ThoriaHamdy99 committed:
add all icons from the design system

mohamedd-hassan committed:
fixed duplicate string

nabilBouzineDev committed:
add a Github Action workflow to ensure project build successfully

nabilBouzineDev committed:
add PR template to ensure detailed description

ThoriaHamdy99 committed:
add another icons from screens

ThoriaHamdy99 committed:
add images for the app with all sizes to fit on all devices.

nabilBouzineDev committed:
move pull_request.yml to .github/workflows package

marah222 committed:
add lightThemeColors

marah222 committed:
modify primaryGradient end/ start

ThoriaHamdy99 committed:
rename image and delete un needed image.

mohamedd-hassan committed:
Improved strings

ThoriaHamdy99 committed:
add missing images.

ThoriaHamdy99 committed:
rename tudee tasks images

ThoriaHamdy99 committed:
add App theme for the project

nabilBouzineDev committed:
update the PR template checklist

ThoriaHamdy99 committed:
rename local colors variable.

Commit:
add usedNumber field to Category and add getTasksByCategoryAndStatus to TaskRepository

Commit:
add edit category functionality to repository

Commit:
add AppSettingsRepository interface to save user data related to the app setting

ThoriaHamdy99 committed:
Merge branch 'develop' of https://github.com/Amsterdam-Team/The-Cute-Tudee into Utrecht-feature/domain

ThoriaHamdy99 commented on app/src/main/java/com/amsterdam/cutetudee/data/repository/TaskRepositoryImpl.kt:
```diff
@@ -33,4 +33,11 @@ class TaskRepositoryImpl : TaskRepository {
         TODO("Not yet implemented")
     }
 
+    override fun getTasksByCategoryAndStatus(
```
can it be getTaskByCategoryIdAndStatus ?

ThoriaHamdy99 commented on app/src/main/java/com/amsterdam/cutetudee/domain/model/Category.kt:
```diff
@@ -7,5 +7,6 @@ import kotlin.uuid.Uuid
     val name: String,
-    val imageUrl: String
+    val imageUrl: String,
+    val usedNumber : Int
```
what is the used number?

ThoriaHamdy99 replied:
If you mean the number of tasks you can name it to numberOfTasks

Noor-Serry replied:
the number of tasks where that category is used

Commit:
refactor: Rename category field and task repository method

nabilBouzineDev commented on app/src/main/java/com/amsterdam/cutetudee/domain/model/Theme.kt:
```diff
@@ -0,0 +1,5 @@
+package com.amsterdam.cutetudee.domain.model
+
+enum class Theme {
```
Could you change this to `Mode` instead of `Them` to avoid conflicting the name with Theme in the UI

Commit:
refactor: Rename Theme to Mode in AppSettingsRepository

This commit renames the `Theme` enum and related methods in the `AppSettingsRepository` interface to `Mode`.

nabilBouzineDev commented on app/src/main/java/com/amsterdam/cutetudee/domain/repository/CategoryRepository.kt:
```diff
@@ -9,5 +9,6 @@ import kotlin.uuid.Uuid
 interface CategoryRepository {
```
We agreed that we should use Service instead of Repository as the abstraction between Data layer and View Model

Noor-Serry replied:
You asked me about that during the meeting just 50 minutes ago, and I mentioned that we can address it after we complete the project by renaming the files. You did not object

nabilBouzineDev replied:
Ok, I got it but I thought it's better to do it now.

Noor-Serry replied:
there are a lot of PRs open now that may effect them

Abanoub-Nagy replied:
I see and agree with @nabilBouzineDev use Service instead of Repository from now if we delay it and start work after component end and start implement the screen and view model the refactor rename will be more large and affect in all project file

eslam-mohamed-dev commented on app/src/main/java/com/amsterdam/cutetudee/domain/model/Mode.kt:
```diff
@@ -0,0 +1,5 @@
+package com.amsterdam.cutetudee.domain.model
+
+enum class Mode {
```
I think ThemeMode is more descriptive, right?

Abanoub-Nagy commented on app/src/main/java/com/amsterdam/cutetudee/domain/model/Mode.kt:
```diff
@@ -0,0 +1,5 @@
+package com.amsterdam.cutetudee.domain.model
+
+enum class Mode {
+    LIGHT,DARk
```
why DARk end with small word should all the char be Capitalize

Commit:
refactor: Rename Mode enum to ThemeMode

Commit:
Refactor: Rename repository interfaces and implementations to services
